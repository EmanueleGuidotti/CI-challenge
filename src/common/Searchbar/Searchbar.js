import { Input } from "antd";
import PropTypes from "prop-types";

const Searchbar = ({ onKeyUp }) => {
  return (
    <Input placeholder="Search..." onKeyUp={(e) => onKeyUp(e.target.value)} />
  );
};

Searchbar.propTypes = {
  onKeyUp: PropTypes.func.isRequired,
};

export { Searchbar };
