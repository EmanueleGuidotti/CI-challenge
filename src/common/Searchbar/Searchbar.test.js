import React from "react";
import renderer from "react-test-renderer";
import { Searchbar } from "./Searchbar";

test("Render Searchbar component properly", () => {
  const onKeyUp = jest.fn();
  const component = renderer.create(<Searchbar onKeyUp={onKeyUp} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Call onKeyUp properly", () => {
  const onKeyUp = jest.fn();
  const component = renderer.create(<Searchbar onKeyUp={onKeyUp} />);
  let tree = component.toJSON();
  tree.props.onKeyUp({ target: { value: "e" } });
  expect(onKeyUp).toBeCalled();
  expect(onKeyUp).toBeCalledTimes(1);
});
