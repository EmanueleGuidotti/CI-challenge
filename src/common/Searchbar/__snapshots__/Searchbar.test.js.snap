// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Searchbar component properly 1`] = `
<input
  className="ant-input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  placeholder="Search..."
  type="text"
  value=""
/>
`;
